### DISCUSSION TITLE
Creating numpy array from JS

### DISCUSSION THREAD

**Discussion by karray** [pyodide/pyodide#3143] [Q&A] [answered]
In earlier version (prior v0.17.0), it was possible to create a `numpy.ndarray` directly from JS like this:

```js
let x = pyodide.globals.np.ones(new Int32Array([3, 3]))
// x >>> [Float64Array(3), Float64Array(3), Float64Array(3)]
```

by creating an array now,

```js
pyodide.globals.get('np').ones(new Int32Array([3, 3]))
```

I get 

```bash
Uncaught PythonError: Traceback (most recent call last):
  File "/lib/python3.10/site-packages/numpy/core/numeric.py", line 204, in ones
    a = empty(shape, dtype, order)
TypeError: expected sequence object with len >= 0 or a single integer
```

Do I need to apply  `to_py` as described in the [documentation](https://pyodide.org/en/stable/usage/type-conversions.html#buffers) (which is only possible within the Python scope) or how can I pass tuple from JS?

**Comment by hoodmane** [comment, marked as answer]
> Do I need to apply `to_py`?

Apparently yes. It is also possible from JavaScript via `pyodide.toPy`:
```
pyodide.globals.get('np').ones(pyodide.toPy([3,3]))
```

I think merging #2906 will fix it so that `pyodide.pyimport("numpy").ones([3,3])` works directly.
